src/ServiceLayer.Mesh/Storage/MeshFilesBlobStore.cs

```diff
@@ -5,9 +5,17 @@ namespace ServiceLayer.Mesh.Storage;
 
 public class MeshFilesBlobStore(BlobContainerClient blobContainerClient) : IMeshFilesBlobStore
 {
-    public Task<Stream> DownloadAsync(MeshFile file)
+    public async Task<Stream> DownloadAsync(MeshFile file)
     {
-        throw new NotImplementedException();
+        var blobClient = blobContainerClient.GetBlobClient(file.BlobPath);
+
+        var dataStream = new MemoryStream();
+
+        await blobClient.DownloadToAsync(dataStream);
+
+        dataStream.Close();
```

Comment:
If we close the stream, the consuming class can't make use of it...

Pretty sure we can do away with the MemoryStream altogether and simplify this to

```
public async Task<Stream> DownloadAsync(MeshFile file)
{
    var blobClient = blobContainerClient.GetBlobClient(file.BlobPath);
    return (await blobClient.DownloadAsync()).Value.Content;
}
```